internal/pkg/describe/backend_app.go

```diff
@@ -0,0 +1,215 @@
+		enableResources: false,
+		store:           svc,
+	}
+	opts.initAppDescriber = func(env string) error {
+		d, err := NewAppDescriber(project, env, app)
+		if err != nil {
+			return err
+		}
+		opts.appDescriber = d
+		return nil
+	}
+	return opts, nil
+}
+
+// NewBackendAppDescriberWithResources instantiates a backend application with stack resources.
+func NewBackendAppDescriberWithResources(project, app string) (*BackendAppDescriber, error) {
+	d, err := NewBackendAppDescriber(project, app)
+	if err != nil {
+		return nil, err
+	}
+	d.enableResources = true
+	return d, nil
+}
+
+// Describe returns info of a backend application.
+func (d *BackendAppDescriber) Describe() (HumanJSONStringer, error) {
+	environments, err := d.store.ListEnvironments(d.app.Project)
+	if err != nil {
+		return nil, fmt.Errorf("list environments: %w", err)
+	}
+
+	var configs []*AppConfig
+	var services []*ServiceDiscovery
+	var envVars []*EnvVars
+	for _, env := range environments {
+		err := d.initAppDescriber(env.Name)
+		if err != nil {
+			return nil, err
+		}
+		appParams, err := d.appDescriber.Params()
+		if err == nil {
+			services = appendServiceDiscovery(services, serviceDiscovery{
+				AppName:     d.app.Name,
+				Port:        appParams[stack.LBWebAppContainerPortParamKey],
+				ProjectName: d.app.Project,
+			}, env.Name)
+			configs = append(configs, &AppConfig{
+				Environment: env.Name,
+				Port:        appParams[stack.LBWebAppContainerPortParamKey],
+				Tasks:       appParams[stack.AppTaskCountParamKey],
+				CPU:         appParams[stack.AppTaskCPUParamKey],
+				Memory:      appParams[stack.AppTaskMemoryParamKey],
+			})
+			backendAppEnvVars, err := d.appDescriber.EnvVars()
+			if err != nil {
+				return nil, fmt.Errorf("retrieve environment variables: %w", err)
+			}
+			envVars = append(envVars, flattenEnvVars(env.Name, backendAppEnvVars)...)
+			continue
+		}
+		if !IsStackNotExistsErr(err) {
+			return nil, fmt.Errorf("retrieve application deployment configuration: %w", err)
+		}
+	}
+	sort.SliceStable(envVars, func(i, j int) bool { return envVars[i].Environment < envVars[j].Environment })
+	sort.SliceStable(envVars, func(i, j int) bool { return envVars[i].Name < envVars[j].Name })
+
+	resources := make(map[string][]*CfnResource)
+	if d.enableResources {
+		for _, env := range environments {
+			stackResources, err := d.appDescriber.AppStackResources()
+			if err == nil {
+				resources[env.Name] = flattenResources(stackResources)
+				continue
+			}
+			if !IsStackNotExistsErr(err) {
+				return nil, fmt.Errorf("retrieve application resources: %w", err)
+			}
+		}
+	}
+
+	return &BackendAppDesc{
+		AppName:          d.app.Name,
+		Type:             d.app.Type,
+		Project:          d.app.Project,
+		Configurations:   configs,
+		ServiceDiscovery: services,
+		Variables:        envVars,
+		Resources:        resources,
+	}, nil
+}
+
+// BackendAppDesc contains serialized parameters for a backend application.
+type BackendAppDesc struct {
```

Comment:
can be private now

Reply:
both `BackendAppDesc` and `WebAppDesc` are used in app show test.

Reply:
oh gotcha, if you want to simplify your life in the `app show` tests you can implement a dummy struct:
```go
type mockDescribeData struct {
    data string
    err error
}

func (m *mockDescribeData) HumanString() string {
   return m.data
}

func (m *mockDescribeData) JSONString() (string, error) {
   return m.data, m.err
}
```

This way you don't need to expose the additional structs and don't have to fill in as much data in `app show` tests